server/remark-includes.ts

```diff
@@ -44,6 +43,56 @@ export interface ParameterAssignments {
   [index: string]: string;
 }
 
+// parseAssignments takes a string containing parameter assignments and parses
+// them into a ParameterAssignments map, making it possible to use these
+// assignments in various operations.
+//
+// It expects the input expression to have a format similar to the following:
+//
+// myvar="myval" myvar2="myval" myvar3="myval"
+//
+// Values must be wrapped in double quotes. Assignments must be separated by
+// single spaces. An equals sign must separate each parameter and its value.
+export function parseAssignments(
+  assignmentSection: string
+): ParameterAssignments {
+  // no parameter assignments to parse
+  if (assignmentSection.length == 0) {
+    return {};
+  }
+
+  if (assignmentSection[assignmentSection.length - 1] !== `"`) {
+    return {};
+  }
+
+  let assignments: ParameterAssignments = {};
+
+  // Since every assignment must end with a double quote and a space, split
+  // assignments on spaces that follow a non-escaped double quote.
+  const rawAssignments = assignmentSection.split(
+    new RegExp('(?<=[^\\\\]") ', "g")
+  );
+
+  const assignmentRegExp = new RegExp('([^ "]+)=(".*")', "g");
```

Comment:
nit: regex will be the bane of me. If we have a complicated-ish regex such as this, can we always leave a comment above it trying to explain it in plain english (as best as possible)